subsys/bluetooth/host/gatt.c

```diff
@@ -2054,18 +2068,12 @@ int bt_gatt_notify_cb(struct bt_conn *conn,
 
 	/* Lookup UUID if it was given */
 	if (params->uuid) {
-		attr = NULL;
-
-		bt_gatt_foreach_attr_type(handle, 0xffff, params->uuid,
-					  NULL, 1, match_uuid, &attr);
+		attr = gatt_find_by_uuid(&tmp, handle, params->uuid);
```

Comment:
It's a static function, so perhaps it's not so bad, but having an API where you provide a pointer to temporary value to a function, without using the result of said pointer, seems somewhat odd. 

But I guess that in this case, `attr == &tmp`, right? So I guess the `attr` parameter to `gatt_find_by_uuid` is basically an `[out]` parameter, and the return value is basically just a nice-to-have thing.